tests/test_func/good/closure_conversion.catala_en

```diff
@@ -19,33 +19,41 @@ type S_in = { x_in: eoption bool; }
 
 type S = { z: eoption integer; }
  
-let topval closure_f : (closure_env, integer) → eoption integer =
+let topval closure_f : (closure_env, integer) → eoption any =
```

Comment:
The Lcalc without exceptions should have typechecking able to infer everything, this is a regression (because of the mix-and-match of `HandleDefaultOpt` I guess). We should investigate to solve this in this PR.